src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverChatController.java

```diff
@@ -0,0 +1,150 @@
+    ) {
+        Pageable pageable = PageRequest.of(0, limit);
+        return driverChatRepository.findAllByOrderByTimeStampDesc(pageable);
+    }
+
+    @ApiOperation(value = "Get a single chat message")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN','ROLE_DRIVER')")
+    @GetMapping("")
+    public DriverChat getMessageById(@ApiParam(name = "id", required = true, example="15", value="id of the message", type="Long") @RequestParam Long id) {
+        DriverChat message =  driverChatRepository.findById(id)
+        .orElseThrow(()->new EntityNotFoundException(DriverChat.class, id));
+
+        return message;
+    }
+
+    @ApiOperation(value="Create a new chat message")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN','ROLE_DRIVER')")
+    @PostMapping("/post")
+    public DriverChat postNewMessage(
+        @RequestBody String messageContent
+    )  throws JsonProcessingException {
+
+        User currentUser = getCurrentUser().getUser();
+        DriverChat chatMessage = DriverChat.builder()
+        .sender(currentUser)
+        .messageContent(messageContent)
+        .timeStamp(LocalDateTime.now())
+        .build();
+      
+        return  driverChatRepository.save(chatMessage);
+    }
+
+    @ApiOperation(value="Update a driver's message")
+    @PreAuthorize("hasRole('ROLE_DRIVER')")
+    @PutMapping("")
+    public DriverChat updateMessageByDriver(
+        @ApiParam(name = "id", required = true, example="15", value="id of the message", type="Long") @RequestParam Long id,
+        @ApiParam(name = "messageContent", required = true, example="can you pick me up?", value="content of new the message", type="String") @RequestParam String messageContent
+    ) {
+        User currentUser = getCurrentUser().getUser();
+        DriverChat message = driverChatRepository.findByIdAndSender(id, currentUser)
+                    .orElseThrow(()->new EntityNotFoundException(DriverChat.class, id));
+        
+        message.setMessageContent(messageContent);
+        driverChatRepository.save(message);
+        return message;
+    }
+
+    @ApiOperation(value="Update a driver's message(admin can edit anyone's messages)")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("admin")
+    public DriverChat updateMessageByAdmin(
+        @ApiParam(name = "id", required = true, example="15", value="id of the message", type="Long") @RequestParam Long id,
+        @ApiParam(name = "messageContent", required = true, example="can you pick me up?", value="content of the message", type="String") @RequestParam String messageContent
+    ) {
+        DriverChat message = driverChatRepository.findById(id)
+                .orElseThrow(()->new EntityNotFoundException(DriverChat.class, id));
+        
+        message.setMessageContent(messageContent);
+        driverChatRepository.save(message);
+        return message;
+    }
+
+    @ApiOperation(value="Delete a chat message if it belongs to you")
+    @PreAuthorize("hasRole('ROLE_DRIVER')")
```

Comment:
I think we'll want this too.

```suggestion
     @PreAuthorize("hasAnyRole('ROLE_DRIVER', 'ROLE_ADMIN')")
```